pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControl.java

```diff
@@ -64,4 +67,36 @@ default boolean hasAccess(String tableName, AccessType accessType, HttpHeaders h
   default boolean hasAccess(AccessType accessType, HttpHeaders httpHeaders, String endpointUrl) {
     return true;
   }
+
+  /**
+   * Return workflow info for authenticating users. Not all workflows may be supported by the pinot UI implementation.
+   *
+   * @return workflow info for user authentication
+   */
+  default AuthWorkflowInfo getAuthWorkflowInfo() {
+    return new AuthWorkflowInfo(WORKFLOW_NONE);
+  }
+
+  /**
+   * Container for authentication workflow info for the Pinot UI. May be extended by implementations.
+   *
+   * Auth workflow info hold any configuration necessary to execute a UI workflow. We currently foresee supporting NONE
+   * (auth disabled), BASIC (basic auth with username and password), and OAUTH2 (token-based workflow via external
+   * issuer)
+   */
+  class AuthWorkflowInfo {
```

Comment:
At LinkedIn, we're using both token-based & certificate-based authentication. We don't have a UI for the endpoints and I'm not sure what kind of information is needed to be returned to UI other than the type of authentication. What do you have in mind?

Reply:
afaik LinkedIn doesn't authenticate web UI calls at all (it's not yet supported by the controller UI). cert-based auth takes place in restli's tls layer, while queries are further authenticated with tokens at the request layer.

this endpoint enables us to configure different UI auth workflows depending on the scenario. at minimum we need to support NONE to maintain backwards compatibility and BASIC to support fully authenticated clusters with this PR.

Reply:
If there's only NONE and BASIC, AuthWorkflowInfo only returns the type of authentication. I still don't understand what information other than auth-type is going to be returned to UI for token or certificate based authentication. Could you elaborate on that?

Reply:
certificate-based auth usually takes place on the lower layer (TLS/SSL).

This endpoint purely serves to configure UI auth workflows as we broaden support:
- disable auth workflows in the UI if pinot is set up without auth
- enable a user/password workflow for basic auth
- provide issuerUrl (with optional info) for oauth2 workflows
- provide secrets/redirects for Azure AD, kerberos, ... (as it comes up)

Reply:
@sajjad-moradi  are you ok with this resolution?

Reply:
Yes sounds good.